### PR TITLE
fix: add detection for php evasion attempt (933100 PL1)

### LINKED ISSUE #3616: PHP filter evasion
### Description

I created something that could be called a xml/php polyglot payload, using a PHP label (aka colon ":").

While I think it triggers anomaly scores and a XSS rule, it doesn't trigger a PHP rule.

I'm not sure if this is fine or not, but wanted to raise it.

### How to reproduce the misbehavior (-> curl call)

Payload:

`<?xml :echo 1;`

On the CRS sandbox:

http://sandbox.coreruleset.org/?x=%3C?xml%20:echo%201;

But executes fine as PHP:

https://onlinephp.io/?c=%3C%3Fxml%20%3Aecho%201%3B&v=8.2.13

A more complex example would be e.g.:

`<?xml :fputCSV($alreadyOpenFile, array("foo", "bar", "hello", "world"));`

which also doesn't trigger a PHP rule.

### Your Environment

* CRS version (e.g., v3.3.4): 4.0.1-dev
* Paranoia level setting (e.g. PL1) : all

### Confirmation

[X] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

### PR DESCRIPTION
**Pull request by franbuehler**
This PR is the second PR that should resolve #3616.
It adds a new detection for `xml\s*[^a-z]` (ignore case). 
Thank you @floyd-fuh for the false negative report and for providing a detection regex.
This PR also adds some tests: false negative test and uppercase test

### CODE REVIEW

File: regex-assembly/933100.ra [hunk 1]
```diff
@@ -11,6 +11,7 @@
     x[^m]
     xm[^l]
     xml[^\s]
+    xml\s*[^a-z]
     xml$
```
Comment (theseion) [inline regex-assembly/933100.ra:15]: `xml$` produces a submatch of `xml[^\s]`. I'd remove `xml$` and add a comment to `xml[^\s]` that it matchs `xml$` implicitly.

File: regex-assembly/933100.ra [hunk 2]
```diff
@@ -11,6 +11,7 @@
     x[^m]
     xm[^l]
     xml[^\s]
+    xml\s*[^a-z]
```
Comment (theseion) [inline regex-assembly/933100.ra:14]: `xml[^\s]` requires a space to follow `xml`.
```suggestion
    xml\s+[^a-z]
```

Comment (theseion) [inline regex-assembly/933100.ra:14]: Add comment here about what we're actually looking for, which is the label marker `:`: `xml :echo`.

**Review by theseion — CHANGES_REQUESTED** [review]

File: regex-assembly/933100.ra (hunk 1 above)
Comment (franbuehler) [reply regex-assembly/933100.ra:15]: Are you sure that `xml$` matches `xml[^\s]`?
For me it doesn't match. Am I wrong?
https://regex101.com/r/eFMckA/1

File: regex-assembly/933100.ra (hunk 2 above)
Comment (franbuehler) [reply regex-assembly/933100.ra]: Resolved, thank you.

File: regex-assembly/933100.ra (hunk 1 above)
Comment (theseion) [reply regex-assembly/933100.ra:16]: Oh I'm so stupid... 🤦. Of course it doesn't, sorry!

**Review by theseion — APPROVED** [review]

**Comment by floyd-fuh** [comment]
looks good to me
